junit-jupiter-api/src/main/kotlin/org/junit/jupiter/api/Assertions.kt

```diff
@@ -95,8 +95,21 @@ fun assertAll(heading: String?, vararg executables: () -> Unit) =
  * ```
  * @see Assertions.assertThrows
  */
-inline fun <reified T : Throwable> assertThrows(noinline executable: () -> Unit): T =
-    Assertions.assertThrows(T::class.java, Executable(executable))
+inline fun <reified T : Throwable> assertThrows(executable: () -> Unit): T =
+    assertThrows(runCatching(executable))
+
+/**
+ * Example usage:
+ * ```kotlin
+ * val exception = assertThrows<IllegalArgumentException>(runCatching {
+ *     throw IllegalArgumentException("Talk to a duck")
+ * })
+ * assertEquals("Talk to a duck", exception.message)
+ * ```
+ * @see Assertions.assertThrows
+ */
+inline fun <reified T : Throwable> assertThrows(result: Result<*>): T =
```

Comment:
Not sure if benefits of this method enough, to be part of API.

Reply:
Inline functions cannot be called from a non-public API, it would need to be collapsed into the calling method.
Is that what you're asking?

Reply:
I'm asking if it makes sense to expose this method as part of API. If they not needed (I think so), they should be collapsed.

Reply:
Yeah it's not completely necessary, and probably doesn't provide a large amount of benefit